src/main/scala/com/richy2509/templator/model/FreemarkerBuilder.scala

```diff
@@ -0,0 +1,123 @@
+package com.richy2509.templator.model
+
+import java.io._
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths}
+import java.util
+
+import com.richy2509.templator.data.TemplatorFreemarkerConfig
+import com.richy2509.templator.exception.Exception.MyTemplateExceptionHandler
+import com.richy2509.templator.utils.StringUtils
+import com.typesafe.scalalogging.Logger
+import freemarker.cache.TemplateLoader
+import freemarker.template.{Configuration, Template}
+
+/**
+  * Created by richardkoehl on 10/12/2016.
+  */
+object FreemarkerBuilder {
+
+  case class FreemarkerConfig(configuration: Configuration, var globalParams: Object = null) {
+
+    val logger: Logger = Logger.apply(FreemarkerConfig.getClass)
+
+    def withGlobalConfig(params: Object): FreemarkerConfig = {
+      val map = params.asInstanceOf[util.HashMap]
+
+      //TODO
+
+      this
+    }
+
+    def withDirectory(path: String): FreemarkerConfig = {
+
+      if (StringUtils.isBlank(path)) {
+        configuration.setTemplateLoader(new TemplateAbsolutePathLoader)
+      } else {
+        val dirPath = Paths.get(path)
+        if (!Files.exists(dirPath)) {
+          Files.createDirectory(dirPath)
+        }
+        configuration.setDirectoryForTemplateLoading(new File(path))
+      }
+      this
```

Comment:
j'aime pas trop le return this.
Je pense qu'en redefinissant la méthode apply par exemple tu pourrais t'affranchir du this